janusgraph-core/src/main/java/org/janusgraph/graphdb/database/management/ManagementSystem.java

```diff
@@ -870,8 +872,6 @@ public boolean equals(Object oth) {
     }
 
     private void setUpdateTrigger(Callable<Boolean> trigger) {
-        //Make sure the most current is the one set
-        if (updatedTypeTriggers.contains(trigger)) updatedTypeTriggers.remove(trigger);
```

Comment:
I'm not too worried about it; the `ManagementSystem` is something used by users to do ~case-by-case schema mutations, as in the List in question probably won't get that long (i.e. users are not going to be doing enough schema mutations to blow out the stack--especially not in one transaction (and every commit requires a new instantiation of the ManagementSystem)) and the GC should do just fine collecting the entire ManagementSystem object once no longer used. And also note this behavior is exactly the same as before.